src/ngx_http_lua_util.c

```diff
@@ -3880,6 +3888,21 @@ ngx_http_lua_init_vm(lua_State *parent_vm, ngx_cycle_t *cycle,
         lua_pop(L, 2);
     }
 
+    if (lmcf->load_resty_core) {
+        lua_getglobal(L, "require");
+        lua_pushstring(L, "resty.core");
+
+        rc = lua_pcall(L, 1, 1, 0);
+        if (rc != 0) {
+            ngx_log_error(NGX_LOG_ERR, log, 0,
+                          "lua_load_resty_core failed to load the resty.core "
+                          "module from https://github.com/openresty/lua-resty"
+                          "-core; ensure you are using an OpenResty release "
+                          "from https://openresty.org/en/download.html "
+                          "(rc: %i, reason: %s)", rc, lua_tostring(L, -1));
```

Comment:
Does this mean lua-nginx-module can not be used outside of OpenResty bundle anymore? In one of our projects we compile lua-nginx-module manually and also include lua-resty-core and set lua package path correctly. Since this PR Nginx fails when starting with this error. When I turn off `lua_load_resty_core` and do just `require("resty.core")` it works as expected.

Reply:
@ElvinEfendi It still can, just a little bit harder as you found out. If you want to add Lua library search paths to your own nginx build, then just override the value of the macro `LUA_DEFAULT_PATH` to include your own installation paths of lua-resty-core (and lua-resty-lrucache). The OpenResty distribution's build system does exactly that.

Reply:
Hmm, it should also work if the user specify the `lua_package_path` and `lua_package_cpath` directives. @thibaultcha The `lua_code_resty_core` directive does not run after these search path nginx directives?

Reply:
@agentzh the project I'm talking about is https://github.com/kubernetes/ingress-nginx/, and we are actually considering to use OpenResty as base image instead of Nginx.

--

I replicated CI image in this repository and renamed `lua-resty-core` and `lua-resty-lrucache` folders to something else. I then edited `t/161-load-resty-core.t` added `ONLY` to the first test and also added `lua_package_path "/lua-resty-corexxx/lib/?.lua;/lua-resty-lrucachexxx/lib/?.lua;;";` using `--- http_config`.
`/lua-resty-corexxx` and `/lua-resty-lrucachexxx` are what I renamed `lua-resty-core` and `lua-resty-lrucache` folders to. Then running test it succeeds. So that tells me it respects `lua_package_path` in this case. Not sure yet why it does not work in https://github.com/kubernetes/ingress-nginx though. The only difference between the Nginx we use in ingress-nginx and 1.15.8 (also the version I used in the above test: https://github.com/ElvinEfendi/lua-nginx-module/pull/1/files#diff-3254677a7917c6c01f55212f86c57fbfR7) that OpenResty uses is that we use Nginx `1.17.0`.

Reply:
@ElvinEfendi Then it's not something on our side? For OpenResty, then it already includes lua-resty-core or lua-resty-lrucache in its default search paths.

Reply:
@ElvinEfendi For us to support you, you need to prepare a minimal and self-contained example. We are not maintaining that ingress-nginx project anyway, not to mention your own setup.

Reply:
@agentzh all I wanted to clarify was "Does this mean lua-nginx-module can not be used outside of OpenResty bundle anymore?" :) Because the error message implies that.

I co-maintain ingress-nginx project - will look into this further on our side to see what's going on and will also potentially switch to OpenResty as base image for that project if the dependencies we need play well with OpenResty bundle.

Thanks for taking time to look into this though!

Reply:
Well, that's a recommendation. Not a requirement.

Reply:
@ElvinEfendi The `lua_load_resty_core` directive respects the `LUA_PATH` environment variable and the `lua_package_path` directive. In fact, I just pushed a new test asserting this behavior: https://github.com/openresty/lua-nginx-module/commit/4102b82cd0971c93663fe89eabd75b73741c4213

Make sure that you also clone [lua-resty-lrucache](https://github.com/openresty/lua-resty-lrucache) which is itself a dependency of lua-resty-core.
Or maybe you set your Lua path in Lua-land (e.g. `package.path = "...;" .. package.path`)? In which case, that'd be too late for this directive to honor the change.

Providing a minimally reproducible example with error logs may be useful here, but again as @agentzh said, this isn't really an issue this module.

This module can still be used independently but the margin for error is high for most users, hence the warnings and other logs (e.g. lua-resty-core, LuaJIT's fork, etc...) asking to prefer the formal OpenResty releases.

Reply:
Where can I set `package.path`?